assets/src/helpers/detectSwipe.ts

```diff
@@ -61,10 +64,16 @@ const firstTouchPosition = (event: TouchEvent): Position => {
   return { x: screenX, y: screenY }
 }
 
-const diff = (end: Position, start: Position): Position => ({
-  x: end.x - start.x,
-  y: end.y - start.y,
-})
+const diff = (end?: Position, start?: Position): Position => {
+  if (!start || !end) {
```

Comment:
Could this explain https://app.asana.com/0/1112935048846093/1132578944531173 and https://app.asana.com/0/1112935048846093/1132578944533173 ?

When would they be undefined? `handleTouchStart` should set them both. Is it ever not called?

Reply:
I think what's happening is that the vehicle data gets refreshed between touchStart and touchEnd, and the value of start and end gets reset

Reply:
Ah. Should `startingPosition` and `endingPosition` be stored in the VPP state instead of `let` variables, so they don't get forgotten?

Reply:
maybe? i don't know how that would interact with the effect lifecycle. this seems to work fine as-is, as far as i have seen.